game-frontend/src/components/role-switch/RoleSwitchButton.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, useMemo } from 'react';
+import { RoleRequest, Role } from '@darwin/types';
+import Button from '../visual/Button';
+import { useSendMessage } from '../../service/game';
+
+interface Props {
+  roleToRequest: Role;
+}
+
+const RoleSwitchButton: FC<Props> = ({ roleToRequest, children }) => {
+  const sendMessage = useSendMessage();
+
+  const sendRoleRequest = useMemo(
+    () => (newRole: Role): void => {
```

Comment:
Why not `useCallback`?

Reply:
Furthermore, what's the benefit of memoing this? The onClick event listener is re-created upon each rerender anyway.